### PR TITLE
server: added separate routers

### PR DESCRIPTION
**Pull request by soniaabhyankar**

### CODE REVIEW

**Comment by prvnbist** [comment]
Hi @soniaabhyankar, thanks for your contribution. I'll review it and get back to you.

**Comment by soniaabhyankar** [comment]
Hey, thanks! And I'm sorry, I didn't know that issues need to be assigned. I'm new to open source.

File: server/index.js [hunk 1]
```diff
@@ -1,131 +1,21 @@
...
-
-// Get list of all files
-app.get('/files', (req, res) => {
-	return getAllFiles(`${defaultLocation}/snippets`).then(files =>
-		res.send({ files })
-	)
-})
+// Routers
+const labelRouter = require('./routes/label');
+const snippetRouter = require('./routes/snippet');
```
Comment (prvnbist) [inline server/index.js]: Can you rename the imports to just `label` and `snippet`?

File: server/index.js [hunk 2]
```diff
@@ -1,131 +1,21 @@
...
-	fs.readFile(filePath, (err, data) => {
-		if (err)
-			return res.status(404).send({
-				success: false,
-				message: `File ${file} doesn't exist!`
-			})
-		return res.status(200).send({ success: true, file: data.toString() })
-	})
-})
+app.use(labelRouter);
```
Comment (prvnbist) [inline server/index.js]: `app.use('/label',label)`
What if we do it like this?

File: server/index.js [hunk 3]
```diff
@@ -1,131 +1,21 @@
...
+app.use(snippetRouter);
```
Comment (prvnbist) [inline server/index.js]: `app.use('/snippet',snippet)`
And do the same with snippet, this is to keep a standard for when new routes are created for like bookmarks or something.

File: server/routes/label.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const fs = require('fs');
+const rimraf = require('rimraf')
+
+
+const router = new express.Router();
+
+const defaultLocation = '..'
+
+router.post('/createLabel', (req, res) => {
```
Comment (prvnbist) [inline server/routes/label.js]: Update the endpoint here to `/create` in accordance with the `route.use` endpoints that I mentioned above.

File: server/routes/label.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+				message: `Folder ${folder} could not be created!`
+			})
+		return res.send({
+			success: true,
+			message: `Label ${folder} has been created!`
+		})
+	})
+})
+
+router.delete('/deleteLabel', (req, res) => {
```
Comment (prvnbist) [inline server/routes/label.js]: Similarly this endpoint to `/delete`

File: server/routes/snippet.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+			return res.status(404).send({
+				success: false,
+				message: `File ${file} doesn't exist!`
+			})
+		return res.status(200).send({ success: true, file: data.toString() })
+	})
+})
+
+// Create Files
+router.post('/saveSnippet', isFileAttached, (req, res) => {
```
Comment (prvnbist) [inline server/routes/snippet.js]: Same for the route endpoints here as well - `/save`

File: server/routes/snippet.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+				message: `File ${file} doesn't exist!`
+			})
+		return res
+			.status(200)
+			.send({ success: true, message: 'File has been saved!' })
+	})
+})
+
+// Delete Files
+router.delete('/deleteSnippet', (req, res) => {
```
Comment (prvnbist) [inline server/routes/snippet.js]: So this endpoint would be `/delete`.

File: server/routes/snippet.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+				message: `File ${file} doesn't exist!`
+			})
+		return res
+			.status(200)
+			.send({ success: true, message: 'File has been deleted!' })
+	})
+})
+
+// Rename File
+router.post('/renameSnippet', (req, res) => {
```
Comment (prvnbist) [inline server/routes/snippet.js]: And finally this would be `/rename`

**Comment by prvnbist** [comment]
Acc. to these the `fetch` endpoints in the React would also have to be updated. With these changes I'll be to keep the endpoint naming consistent across routes. Let me know if you have any question or suggestion for the same.

File: server/index.js (hunk 1 above)
Comment (soniaabhyankar) [reply server/index.js]: Yes, will do.

File: server/index.js (hunk 3 above)
Comment (soniaabhyankar) [reply server/index.js]: Yes, sounds good to me. Do you want it to be something like `app.use('/api/snippet',snippet)` ?
Or just `/snippet` ?

**Comment by soniaabhyankar** [comment]
I'll make the above changes, and also change the `fetch `endpoints in React.

**Review by prvnbist — APPROVED** [review]

**Comment by prvnbist** [comment]
Thanks for your contribution🎉

**Pull request merged**